### DISCUSSION TITLE
상속된 제페토 컴포넌트의 변수를 출력 할 수 없습니다.

### DISCUSSION THREAD

**Discussion by kyn320** [naverz/zepeto-studio-kor#631] [5. Q&A]
부모 클래스에 public 변수가 존재하고,
자식 클래스를 제페토 스크립트로 사용하려고 하면,
ArgumentNullExeption이 출력됩니다.

아직 제페토에서는 상속 기반의 컴포넌트 구조내 인스펙터 표시 지원하지 않는걸까요?

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요~ string, number등 타입 변수는 상속 후 인스펙터가 잘 나타납니다.
다만 커스텀 타입의 경우는 인스펙터 지원이 되지 않고 있습니다. 🙏
직접 구현한 코드 및 오류 스크린샷, 월드 버전을 알려주시면 더 정확한 도움을 드릴 수 있을 것 같습니다.

**Reply by kyn320** [reply]
![image](https://user-images.githubusercontent.com/11557205/164413758-26587bab-8542-4719-8240-74aca3f793b5.png)
![image](https://user-images.githubusercontent.com/11557205/164413797-b583e027-98aa-45fb-8a55-86d7b06f4d0f.png)
![image](https://user-images.githubusercontent.com/11557205/164413858-f34eb033-7c7c-4d8f-84a2-a1f950093f0f.png)
![image](https://user-images.githubusercontent.com/11557205/164413896-167db025-63ec-448e-8fbe-d7a85b40101b.png)
![image](https://user-images.githubusercontent.com/11557205/164414431-9c367858-768b-483a-8ee9-fc697ae035a9.png)

버전은 제일 최신인 1.2 입니다.

**Comment by Zepeto-TechEvangelist** [comment]
<img width="429" alt="스크린샷 2022-04-21 오후 5 38 28" src="https://user-images.githubusercontent.com/100118650/164415435-797a9404-8c2e-45a0-b711-eeb0836560f6.png">
<img width="319" alt="스크린샷 2022-04-21 오후 5 38 20" src="https://user-images.githubusercontent.com/100118650/164415452-0c22bf29-381c-4f31-b11d-553bd1d0f9fa.png">

알려주신 내용과 동일한 코드 작성했으나 오류는 나타나지 않았습니다. 🤔

**Reply by kyn320** [reply]
그럼 제가 어떻게 대처 하면 될까요..? 상속 기능은 아무것도 못 사용하고 있어서 불편한네요.

**Reply by Zepeto-TechEvangelist** [reply]
ArgumentNullException 오류에 초점을 맞춰서 오브젝트 하나하나 Inspector에 빠진 부분이 없는지 살펴보시는 것이 좋을 것 같습니다.

**Reply by kyn320** [reply]
다시 시도해보았는데, 리빌드 될 때 수시로 해당 오류가 뜨고있네요..
될때도 있고 안될때도 있는것 같습니다. 인스펙터에 한번 노출 되고 난후에 ts 파일을 수정하면 바로 에러가 뜹니다.
일단 reimport같은 기능으로 우회 가능한지 해봐야겠네요. 감사합니다.
